tests/integration/100.command.dev.test.cjs

```diff
@@ -781,6 +770,53 @@ test('should respect in-source configuration from edge functions', async (t) =>
           handler: () => new Response('Hello world'),
           name: 'hello',
         })
+        .buildAsync()
+
+      await waitForLogMatching('Reloaded edge function')
+
+      const res2 = await got(`http://localhost:${port}/hello-1`, { throwHttpErrors: false })
+
+      t.is(res2.statusCode, 404)
+
+      const res3 = await got(`http://localhost:${port}/hello-2`, { throwHttpErrors: false })
+
+      t.is(res3.statusCode, 200)
+      t.is(res3.body, 'Hello world')
+    })
+  })
+})
+
+test('should respect in-source configuration from internal edge functions', async (t) => {
+  await withSiteBuilder('site-with-internal-edge-functions', async (builder) => {
+    const publicDir = 'public'
+    await builder
+      .withNetlifyToml({
+        config: {
+          build: {
+            publish: publicDir,
+          },
+        },
+      })
+      .withEdgeFunction({
+        config: () => ({ path: '/internal-1' }),
+        handler: () => new Response('Hello from an internal function'),
+        internal: true,
+        name: 'internal',
+      })
```

Comment:
About the test failures: It might be that we do not update the actual routing for edge functions on reload in the edge-functions-proxy. That is what I would check first if that actually works. 
I'm not even sure if that works with a `netlify.toml`, so updating the routing in `netlify.toml` while `netlify dev` is running.

Reply:
Okay here is the watch logic from edge-functions and it looks like we do some special logic for the config file and we probably need to adapt that for isc-edge-functions.